src/test/scala/chiselTests/RecordSpec.scala

```diff
@@ -0,0 +1,45 @@
+// See LICENSE for license details.
+
+package chiselTests
+
+import chisel3._
+import scala.collection.immutable.ListMap
+
+class RecordSpec extends ChiselFlatSpec {
+  class RecordFromMap(val elements: ListMap[String, Data]) extends Record {
+    override def cloneType =
```

Comment:
The particulars of Aggregate cloning may require your cloneType to be calling chiselCloneType. Otherwise, an intermediate Aggregate may fail to get the proper direction.